feature-libs/product-configurator/rulebased/cpq/cpq-configurator-rest.service.ts

```diff
@@ -0,0 +1,235 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { ConverterService } from '@spartacus/core';
+import { forkJoin, Observable, of } from 'rxjs';
+import { map, switchMap } from 'rxjs/operators';
+import { Configurator } from '../core/model/configurator.model';
+import { CPQ_CONFIGURATOR_VIRTUAL_ENDPOINT } from '../root/interceptor/cpq-configurator-rest.interceptor';
+import {
+  CPQ_CONFIGURATOR_NORMALIZER,
+  CPQ_CONFIGURATOR_OVERVIEW_NORMALIZER,
+  CPQ_CONFIGURATOR_QUANTITY_SERIALIZER,
+  CPQ_CONFIGURATOR_SERIALIZER,
+} from './cpq-configurator.converters';
+import { Cpq } from './cpq.models';
+
+@Injectable({ providedIn: 'root' })
+export class CpqConfiguratorRestService {
+  constructor(
+    protected http: HttpClient,
+    protected converterService: ConverterService
+  ) {}
+
+  /**
+   * Will create a new runtime configuration for the given product id
+   * and read this default configuration from the CPQ system.
+   */
+  createConfiguration(
+    productSystemId: string
+  ): Observable<Configurator.Configuration> {
+    return this.callConfigurationInit(productSystemId).pipe(
+      switchMap((configCreatedResponse) => {
+        return this.callConfigurationDisplay(
+          configCreatedResponse.configurationId
+        ).pipe(
+          this.converterService.pipeable(CPQ_CONFIGURATOR_NORMALIZER),
+          map((resultConfiguration) => {
+            return {
+              ...resultConfiguration,
+              configId: configCreatedResponse.configurationId,
+            };
+          })
+        );
+      })
+    );
+  }
+
+  readConfiguration(
+    configId: string,
+    tabId?: string
+  ): Observable<Configurator.Configuration> {
+    return this.callConfigurationDisplay(configId, tabId).pipe(
+      this.converterService.pipeable(CPQ_CONFIGURATOR_NORMALIZER),
+      map((resultConfiguration) => {
+        return {
+          ...resultConfiguration,
+          configId: configId,
+        };
+      })
+    );
+  }
+
+  readConfigurationOverview(
+    configId: string
+  ): Observable<Configurator.Overview> {
+    return this.getConfigurationWithAllTabsAndAttribues(configId).pipe(
```

Comment:
Typo 'Attribues'. Here and in other places

Reply:
Pick!